File: autolink-desktop/src/main/java/models/PartnershipRequest.java
package models;

import java.time.LocalDateTime;

public class PartnershipRequest {
    private int idRequest;
    private Entreprise entreprise;
    private String status;
    private String companyEmail;
    private String phone;
    private String address;
    private String taxCode;
    private String partnershipType;
    private String description;
    private LocalDateTime submittedAt;

    public PartnershipRequest() {}

    public PartnershipRequest(int idRequest, Entreprise entreprise, String status, String companyEmail,
                              String phone, String address, String taxCode, String partnershipType,
                              String description, LocalDateTime submittedAt) {
        this.idRequest = idRequest;
        this.entreprise = entreprise;
        this.status = status;
        this.companyEmail = companyEmail;
        this.phone = phone;
        this.address = address;
        this.taxCode = taxCode;
        this.partnershipType = partnershipType;
        this.description = description;
        this.submittedAt = submittedAt;
    }

    public int getIdRequest() {
        return idRequest;
    }

    public void setIdRequest(int idRequest) {
        this.idRequest = idRequest;
    }

    public Entreprise getEntreprise() {
        return entreprise;
    }

    public void setEntreprise(Entreprise entreprise) {
        this.entreprise = entreprise;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getCompanyEmail() {
        return companyEmail;
    }

    public void setCompanyEmail(String companyEmail) {
        this.companyEmail = companyEmail;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getTaxCode() {
        return taxCode;
    }

    public void setTaxCode(String taxCode) {
        this.taxCode = taxCode;
    }

    public String getPartnershipType() {
        return partnershipType;
    }

    public void setPartnershipType(String partnershipType) {
        this.partnershipType = partnershipType;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public LocalDateTime getSubmittedAt() {
        return submittedAt;
    }

    public void setSubmittedAt(LocalDateTime submittedAt) {
        this.submittedAt = submittedAt;
    }
}

File: src/Entity/PartnershipRequest.php
<?php

namespace App\Entity;

use App\Repository\PartnershipRequestRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: PartnershipRequestRepository::class)]
class PartnershipRequest
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idRequest = null;

    #[ORM\Column(nullable: false)] // Store the user ID directly
    private ?int $entrepriseId = null;

    #[ORM\Column(length: 20)]
    private ?string $status = 'Pending';

    #[ORM\Column(length: 255)]
    #[Assert\Email]
    private ?string $companyEmail = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank]
    private ?string $phone = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank]
    private ?string $address = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank]
    private ?string $taxCode = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank]
    private ?string $partnershipType = null;

    #[ORM\Column(type: Types::TEXT)]
    #[Assert\NotBlank]
    private ?string $description = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $submittedAt = null;

    public function __construct()
    {
        $this->submittedAt = new \DateTime();
        $this->status = 'Pending';
    }

    public function getIdRequest(): ?int
    {
        return $this->idRequest;
    }

    public function getEntrepriseId(): ?int
    {
        return $this->entrepriseId;
    }

    public function setEntrepriseId(int $entrepriseId): static
    {
        $this->entrepriseId = $entrepriseId;
        return $this;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(string $status): static
    {
        $this->status = $status;
        return $this;
    }

    public function getCompanyEmail(): ?string
    {
        return $this->companyEmail;
    }

    public function setCompanyEmail(string $companyEmail): static
    {
        $this->companyEmail = $companyEmail;
        return $this;
    }

    public function getPhone(): ?string
    {
        return $this->phone;
    }

    public function setPhone(string $phone): static
    {
        $this->phone = $phone;
        return $this;
    }

    public function getAddress(): ?string
    {
        return $this->address;
    }

    public function setAddress(string $address): static
    {
        $this->address = $address;
        return $this;
    }

    public function getTaxCode(): ?string
    {
        return $this->taxCode;
    }

    public function setTaxCode(string $taxCode): static
    {
        $this->taxCode = $taxCode;
        return $this;
    }

    public function getPartnershipType(): ?string
    {
        return $this->partnershipType;
    }

    public function setPartnershipType(string $partnershipType): static
    {
        $this->partnershipType = $partnershipType;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;
        return $this;
    }

    public function getSubmittedAt(): ?\DateTimeInterface
    {
        return $this->submittedAt;
    }

    public function setSubmittedAt(\DateTimeInterface $submittedAt): static
    {
        $this->submittedAt = $submittedAt;
        return $this;
    }
}
